components/admin_console/permission_schemes_settings/permission_description.tsx

```diff
@@ -0,0 +1,106 @@
+
+import {generateId} from 'utils/utils';
+
+import {Role} from '@mattermost/types/roles';
+
+import {AdditionalValues} from './permissions_tree/types';
+
+type Props = {
+    intl: IntlShape;
+    id: string;
+    rowType: string;
+    inherited?: Partial<Role>;
+    selectRow: (id: string) => void;
+    additionalValues?: AdditionalValues | AdditionalValues['edit_post'];
+}
+
+const PermissionDescription = (props: Props) => {
+    const [open, setOpen] = useState(false);
+    const randomId = generateId();
+    const contentRef = useRef<HTMLSpanElement>(null);
+
+    const closeTooltip = () => setOpen(false);
+
+    const openTooltip = (e: MouseEvent) => {
+        const elm = e.currentTarget.querySelector('span');
+        const isElipsis = elm ? elm.offsetWidth < elm.scrollWidth : false;
+        setOpen(isElipsis);
+    };
+
+    const parentPermissionClicked = (e: MouseEvent) => {
+        const parent = (e.target as HTMLSpanElement).parentElement;
+        const isInheritLink = parent?.parentElement?.className === 'inherit-link-wrapper';
+        if (parent?.className !== 'permission-description' && !isInheritLink) {
+            e.stopPropagation();
+        } else if (isInheritLink) {
+            props.selectRow(props.id);
+            e.stopPropagation();
+        }
+    };
+
+    const {inherited, id, rowType} = props;
+
+    let content: string | JSX.Element = '';
+    if (inherited) {
+        content = (
+            <span className='inherit-link-wrapper'>
+                <FormattedMarkdownMessage
+                    id='admin.permissions.inherited_from'
+                    values={{
+                        name: props.intl.formatMessage({
+                            id: 'admin.permissions.roles.' + inherited.name + '.name',
+                            defaultMessage: inherited.display_name,
+                        }),
+                    }}
+                />
+            </span>
+        );
+    } else {
+        content = (
+            <FormattedMessage
+                id={'admin.permissions.' + rowType + '.' + id + '.description'}
+                values={props.additionalValues}
+            />
+        );
+    }
+    let tooltip: JSX.Element | null = (
+        <Overlay
+            show={open}
+            placement='top'
+            target={(contentRef.current as HTMLSpanElement)}
+        >
+            <Tooltip id={randomId}>
+                {content}
+            </Tooltip>
+        </Overlay>
+    );
+    if (content.props.values && Object.keys(content.props.values).length > 0) {
+        tooltip = null;
+    }
+    content = (
+        <span
+            className='permission-description'
+            onClick={parentPermissionClicked}
+            ref={contentRef}
+            onMouseOver={openTooltip}
+            onMouseOut={closeTooltip}
+        >
+            {content}
+            {tooltip}
+        </span>
+    );
+
+    return content;
+};
+
+export default injectIntl(PermissionDescription);
```

Comment:
nit: you can use ```useIntl``` hook here as this is a functional component and you don't need to wrap in HOC.

Reply:
Good point. I am not very familiar with `useIntl`. Please check my commits ([1](https://github.com/mattermost/mattermost-webapp/pull/10661/commits/94d44b4701d200628b3509deefb9817fd8f1513f) and [2](https://github.com/mattermost/mattermost-webapp/pull/10661/commits/5b951813436c0c347bcdd09c4ada4be168e59f13)) to see if I missed anything